packages/shared/src/components/modals/SubmitArticle.tsx

```diff
@@ -0,0 +1,224 @@
+      event_name: 'submit article fail',
+      extra: JSON.stringify({
+        reason,
+      }),
+    });
+  };
+
+  const onUrlChanged = () => {
+    if (submitFormRef.current) {
+      if (existingArticle) {
+        setExistingArticle(null);
+      }
+      if (urlHint) {
+        setUrlHint(null);
+      }
+      setEnableSubmission(submitFormRef.current.checkValidity());
+    }
+  };
+
+  const onSubmit = async (
+    event: React.FormEvent<HTMLFormElement>,
+  ): Promise<void> => {
+    event.preventDefault();
+    const data = formToJson<{ articleUrl: string }>(event.currentTarget);
+
+    trackEvent({
+      event_name: 'submit article',
+      feed_item_title: submitArticleModalButton,
+      extra: JSON.stringify({ url: data?.articleUrl }),
+    });
+
+    setIsValidating(true);
+
+    if (!data.articleUrl) {
+      setUrlHint('Please submit a valid URL');
+      setIsValidating(false);
+      return;
+    }
+
+    const defaultErrorMessage = 'Something went wrong, try again';
+    try {
+      const res = await submitArticle(data.articleUrl);
+      const {
+        reason = defaultErrorMessage,
+        post,
+        submission,
+      } = res.submitArticle;
+      if (submission) {
+        setIsSubmitted(true);
+        trackEvent({ event_name: 'submit article succeed' });
+      } else if (post) {
+        setExistingArticle({ post });
+        submitArticleFailEvent('Article exists already');
+      } else if (reason) {
+        setUrlHint(reason ?? defaultErrorMessage);
+        submitArticleFailEvent(reason);
+      }
+    } catch (err) {
+      setUrlHint(defaultErrorMessage);
+      setEnableSubmission(false);
+      submitArticleFailEvent(defaultErrorMessage);
+    } finally {
+      setIsValidating(false);
+    }
+  };
+
+  const getButton = (): ReactElement => {
+    if (isSubmitted) {
+      return (
+        <>
+          <div className="flex items-center mt-4 mb-6 h-8 font-bold rounded-10 w-fit bg-theme-status-success text-theme-label-invert padding px-[0.9375rem]">
+            Request sent
+          </div>
+          <p className="mb-2 typo-callout text-theme-label-tertiary">
+            You will be notified via email about the article request status.
+          </p>
+        </>
+      );
+    }
+
+    if (isEnabled) {
+      return (
+        <Button
+          className="mt-4 btn-primary "
+          buttonSize="small"
+          type="submit"
+          aria-label={submitArticleModalButton}
+          disabled={!enableSubmission}
+          loading={isValidating}
+        >
+          <span className={isValidating && 'invisible'}>
+            {submitArticleModalButton}
+          </span>
+        </Button>
+      );
+    }
+    return (
+      <Button
+        tag="a"
+        href={requestAccessSubmitArticle}
+        target="_blank"
+        rel="noopener"
+        aria-label="Request access"
+        className="mt-4 w-fit btn-secondary"
+      >
+        Request access
+      </Button>
+    );
```

Comment:
Not sure I want this part anymore from a product perspective. I'll talk to Tsahi about it

Reply:
This is gone :)